### PR TITLE
RadialBar: activeShape should work with Tooltip

### PR DESCRIPTION
**Pull request by andrewangelle**
The existing activeShape prop on `<RadialBar />` should work with the `<Tooltip />` in the same manner as `<Line />`, `<Bar />`, `<Funnel/>` do.
## Description

- #### src/chart/generateCategoricalChart.tsx
  - Added another if block to render the graphical item when `activeShape` is truthy.
Inside the new if block, render the graphical item and pass in `activeIndex` which defaults to user defined `activeIndex` and falls back to `activeTooltipIndex` when undefined.

- #### src/polar/RadialBar.tsx
  - Renamed the generically exported type named Props to RadialBarProps
  - Refactored static render method out to RadialBarUtils
  - Updated type of `RadialBar.activeShape` to match the signature of activeLine/activeBar/activeShape in Line/Bar/Funnel components

## Related Issue
#3792 

## Motivation and Context

To provide a consistent interface between Line / Bar / Funnel / RadialBar and their respective props of activeDot / activeBar / activeShape

## How Has This Been Tested?
Unit tests and storybook

## Screenshots (if appropriate):
Screenshot 2023-09-29 at 5 23 19 PM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/polar/RadialBar.tsx [hunk 1]
```diff
@@ -39,18 +41,17 @@ type RadialBarDataItem = SectorProps & {
   background?: SectorProps;
 };
 
-type RadialBarShape = ReactElement | ((props: Props) => ReactNode);
-type RadialBarBackground = ReactElement | ((props: Props) => ReactNode) | SectorProps | boolean;
+type RadialBarBackground = ActiveShape<SectorProps>;
```
Comment (nikolasrieble) [inline src/polar/RadialBar.tsx:44]: Nice.

File: src/polar/RadialBar.tsx [hunk 2]
```diff
@@ -366,8 +356,8 @@ export class RadialBar extends PureComponent<Props, State> {
         return null;
       }
 
-      const props = {
-        cornerRadius,
+      const props: RadialBarSectorProps = {
+        cornerRadius: typeof cornerRadius === 'string' ? parseInt(cornerRadius, 10) : cornerRadius,
```
Comment (nikolasrieble) [inline src/polar/RadialBar.tsx:360]: Nit: We could extract `parseCornerRadius`

File: src/chart/generateCategoricalChart.tsx [hunk 3]
```diff
@@ -2151,6 +2151,12 @@ export const generateCategoricalChart = ({
             return [graphicalItem, this.renderActiveBar({ item, childIndex: activeTooltipIndex })];
           }
 
+          if (activeShape) {
+            const activeIndex =
+              element.props.activeIndex !== undefined ? element.props.activeIndex : activeTooltipIndex;
+            return [cloneElement(element, { ...item.props, ...itemEvents, activeIndex }), null, null];
```
Comment (nikolasrieble) [inline src/chart/generateCategoricalChart.tsx:2157]: Just wondering, do you see a reason why activeBar should be treated separately? Could we not treat all activeShapes the same? 

Or is there some specific behaviour the I am not aware of?

File: src/util/RadialBarUtils.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  if (isValidElement(option)) {
+    sectorShape = cloneElement(option, props);
+  } else if (_.isFunction(option)) {
+    sectorShape = option(props);
+  } else if (_.isPlainObject(option) && !_.isBoolean(option)) {
+    const elementProps = typeGuardSectorProps(option, props);
+    sectorShape = <Sector {...elementProps} />;
+  } else {
+    sectorShape = <Sector {...props} />;
+  }
```
Comment (nikolasrieble) [inline src/util/RadialBarUtils.tsx:42]: I have seen this logic before in various variants. Should we possibly extract this as a helper and reuse? 

Screenshot 2023-10-01 at 12 14 46

File: src/util/RadialBarUtils.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  if (props.isActive) {
+    return <Layer className="recharts-active-shape">{sectorShape}</Layer>;
```
Comment (nikolasrieble) [inline src/util/RadialBarUtils.tsx:45]: Not sure about best practice, but we are already using this twice so far, we could extract it as a constant. 

What is the use case of this className? Testing? Enabling the user to further adapt the style? 

If the reason is testing, should we then not be using data-testid attribute instead?

File: storybook/stories/API/chart/RadialBarChart.stories.tsx [hunk 6]
```diff
@@ -1,30 +1,55 @@
...
   argTypes: {
     ...CategoricalChartProps,
+    activeIndex: {
+      description:
+        'The index of the individual shapes that make up the Funnel to be marked as active, and render props.activeShape as a result',
+      table: {
+        type: {
+          summary: 'number | undefined',
+        },
+        defaultValue: undefined,
```
Comment (nikolasrieble) [inline storybook/stories/API/chart/RadialBarChart.stories.tsx:19]: Do we need to define these here? Do they not get auto-filled from the code? 🤔

**Review by nikolasrieble — APPROVED** [review]
Overall great addition! I have a few comments and questions but nothing blocking. Thank you for your (repeated ❤️ ) contribution

File: src/util/RadialBarUtils.tsx (hunk 5 above)
Comment (andrewangelle) [reply src/util/RadialBarUtils.tsx]: > What is the use case of this className? Testing? Enabling the user to further adapt the style?

I believe if the user provides `activeShape` an object of attributes, or as a boolean, then they could use this classname to target styling.

File: storybook/stories/API/chart/RadialBarChart.stories.tsx (hunk 6 above)
Comment (andrewangelle) [reply storybook/stories/API/chart/RadialBarChart.stories.tsx]: Should be fine. I removed these.

File: src/util/RadialBarUtils.tsx (hunk 4 above)
Comment (andrewangelle) [reply src/util/RadialBarUtils.tsx]: Addressed here: https://github.com/recharts/recharts/pull/3803/commits/d28ad574c21165a70e11697e70a9e753a7d063c3

Let me know if you have any feedback on that refactor.

File: src/chart/generateCategoricalChart.tsx (hunk 3 above)
Comment (andrewangelle) [reply src/chart/generateCategoricalChart.tsx:2118]: Addressed here: https://github.com/recharts/recharts/pull/3803/commits/80bd132bfecaee13a84d09b661a86a59c425a0eb

**Comment by nikolasrieble** [comment]
Thank you @andrewangelle! I tested it again and found no issue.

**Pull request merged**